docs/backend/eval-repo.md

```diff
@@ -0,0 +1,44 @@
+# New Repository Proposal: eval
+
+## Summary
+
+This document proposes a new repository under the `instructlab` GitHub organization:
+
+- `instructlab/eval`
+
+## Background
+
+The `instructlab/instructlab` repository currently includes no real implementation
+of Evaluation as described by the [LAB paper](https://arxiv.org/abs/2403.01081). The
+closest implementation currently in `instructlab/instructlab` via the `ilab test` command.
+
+`ilab test` as of this writing is only implemented for macOS with M-series chips. It uses
+a JSON Lines file and a LoRA adapter to compare output of a given model before and after
+LoRA training with MLX, thus the macOS M-series dependency.
+
+We desire to build out an implementation closer to the described evaluation in the paper,
+using more high-level evaluation schemes such as
+[Massive Multitask Language Understanding](https://arxiv.org/abs/2009.03300) (MMLU).
+We propose a new repository to house this code that publishes a new Python library
+called `instructlab-eval`. The reasoning for a new repository and library includes:
+
+- We expect multiple consumers of this code. The `ilab` CLI is one, but we also envision
+building a REST API around it to help support scaling out this functionality on a cluster.
+- We expect there is broader community interest in an open-source library and service for
+evaluation. We envision this library could support other evaluation techniques over time.
```

Comment:
Can you add something like ...

- We also realize that much of model evaluation is generally useful outside the context of InstructLab. Other libraries may emerge in the broader ecosystem that handle parts of what we need, while this library will always remain to handle the InstructLab-specific details of how evaluation works in our workflow.

Reply:
I like it!